Sorting/RadixSort.cpp

```diff
@@ -0,0 +1,120 @@
+/*
+    Radix sort
+    ----------
+    An efficient, key-index-based, comparision based sorting algorithm.
+    It sorts based on the keys obtained on each index of the numbers to be sorteed.
+*/
+
+#include <iostream>
+#include <vector>
+#include "SortingUtils.hpp"
+
+using namespace std;
+
+// The setFactors function sets the factors which depend on whether the sorting is needed to be done in ascending or descending order
+void setFactors(const int order, int& mult_factor, int& add_factor) {
+    if(order == -1) {
+        mult_factor = -1;
+        add_factor = 9;
+    }
+}
+
+/*
+    count_sort
+    ----------
+    Performs Count Sort on the elements passed by the radix_sort function
+
+    Time complexity
+    ---------------
+    O(N + B), where N is the number of keys and B is the base for representing the numbers(10 in case of decimal system)
+
+    Space complexity
+    ----------------
+    O(N + B), where N is the number of keys and B is the base for representing the numbers(10 in case of decimal system)
+    This space is used for count sorting each key and temporarily creating a sorted array.
+*/
+void count_sort(vector<int>& values, const size_t size, const int I, const int order, const bool to_show_state, const int mult_factor, const int add_factor) {
+    int counter[10] = {0};
+    int* output = (int*)malloc(size*sizeof(int));
+    for(int i = 0; i < size; i++)
+        counter[add_factor + (mult_factor * ((values[i]/I) % 10))]++;
+    for(int i = 1; i < 10; i++)
+        counter[i] += counter[i - 1];
+    for(int i = size - 1; i >= 0; i--) {
+        output[counter[add_factor + (mult_factor * ((values[i]/I) % 10))] - 1] = values[i];
+        counter[add_factor + (mult_factor * ((values[i]/I) % 10))]--;
```

Comment:
Since the value `add_factor + (mult_factor * ((values[i]/I) % 10))` depends on `values[i]`, or more specifically, just the index `i`, you cannot store it in a variable. You would either need to pass the value `values[i]`, or pass `values` and `i`, along with `add_factor`, `mult_factor` and `extractor` to a function which would return the required value.
As to why use a function for this at all, it is because it would reduce code repetition and make the code shorter and simpler to understand.
**NOTE:** Instead of creating a global function and passing it all the values (`values[i]`, `add_factor`, `mult_factor` and `extractor`), you can use a [lambda expression](http://en.cppreference.com/w/cpp/language/lambda) and pass only `values[i]` as the parameter to further simplify the expression.